pkg/controllers/state/cluster.go

```diff
@@ -262,13 +262,20 @@ func (c *Cluster) UpdateNode(ctx context.Context, node *v1.Node) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
+	managed := node.Labels[v1alpha5.ProvisionerNameLabelKey] != "" || node.Labels[v1beta1.NodePoolLabelKey] != ""
+	initialized := node.Labels[v1beta1.NodeInitializedLabelKey] != ""
 	if node.Spec.ProviderID == "" {
 		// If we know that we own this node, we shouldn't allow the providerID to be empty
-		if node.Labels[v1alpha5.ProvisionerNameLabelKey] != "" || (node.Labels[v1beta1.NodePoolLabelKey] != "") {
+		if managed {
```

Comment:
Shouldn't this be `!managed`?

Reply:
No; this means that if we don't have the providerID and are managing the node, that we should reject updates right now since we expect there to be a providerID and if we allowed the update to go through, we could get a "ghost node" that should otherwise be matched up with a Machine.